signature/src/bls_over_bn254.rs

```diff
@@ -224,24 +225,60 @@ impl AggregateableSignatureSchemes for BLSOverBN254CurveSignatureScheme {
 // =====================================================
 // Signing key
 // =====================================================
-#[tagged(tag::BLS_SIGNING_KEY)]
-#[derive(
-    Clone,
-    Hash,
-    Default,
-    Zeroize,
-    Eq,
-    PartialEq,
-    CanonicalSerialize,
-    CanonicalDeserialize,
-    Derivative,
-    Ord,
-    PartialOrd,
-)]
+#[derive(Clone, Hash, Default, Zeroize, Eq, PartialEq)]
 #[zeroize(drop)]
-#[derivative(Debug)]
 /// Signing key for BLS signature.
-pub struct SignKey(#[derivative(Debug = "ignore")] pub(crate) ScalarField);
+pub struct SignKey(pub(crate) ScalarField);
+
+impl core::fmt::Debug for SignKey {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        f.debug_struct("<private signing key>").finish()
+    }
+}
+
+impl SignKey {
+    /// Signature Key generation function
+    pub fn generate<R: CryptoRng + RngCore>(prng: &mut R) -> SignKey {
+        SignKey(ScalarField::rand(prng))
+    }
+
+    /// Explicit calls to serialize a `SignKey` into bytes.
+    pub fn to_bytes(&self) -> Vec<u8> {
+        self.0.into_bigint().to_bytes_le()
+    }
+
+    /// Deserialize `SignKey` from bytes.
+    pub fn from_bytes(bytes: &[u8]) -> Self {
+        Self(ScalarField::from_le_bytes_mod_order(bytes))
+    }
+
+    /// Explicit calls to serialize a `SignKey` into `TaggedBase64`.
+    pub fn to_tagged_base64(&self) -> Result<TaggedBase64, Tb64Error> {
+        TaggedBase64::new(tag::BLS_SIGNING_KEY, &self.to_bytes())
+    }
```

Comment:
1. should we explicit about `tag::BLS_OVER_BN254_SIGN_KEY`?
2. should we add a feature request on `tagged` procedure macro that has a mode that instead of implement serde::Serialize for those that impl CanonicalSerde, but auto derive serde for those that impl `from/to_bytes()`? (cc @jbearer)

Reply:
🤔 
on second thought, instead of `to_tagged_base64` and `TryFrom<TaggedBase64>`, we should skip this and manually implement `serde::Serialize` for this types (that internal call `from/to_bytes()`, 
wdyt?